File: c/functionTest.c
#include <stdio.h>

void test(){
	void f(){
		printf("%s", "apple");
	}
	f();
}

void test2(){
	int f(void (*p)(char* words)){
		p("banana");
	}
	void f2(char* words){
		printf("%s", words);
	}
	f(f2);
}

void main(){
	test2();
}

File: rust/test2/src/functionTest.rs

pub fn test(){
	fn f(x:i32, y:i32) -> i32 {
		return x+y;
	}
	let a = f(3, 5);
	println!("{}", a);
}

pub fn test2(){
	fn f(x:i32, y:i32) -> i32 {
		x+y
	}
	let a = f(3, 5);
	println!("{}", a);
}
